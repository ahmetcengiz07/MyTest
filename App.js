/* eslint-disable react-native/no-inline-styles */
import React, {useState} from 'react';
import {
  SafeAreaView,
  Text,
  TextInput,
  StyleSheet,
  FlatList,
  View,
} from 'react-native';
import Button from './Src/Button';

const App = () => {
  const [list, setList] = useState([]);
  const [text, setText] = useState('');

  const renderList = ({item}) => {
    return (
      <View
        style={[
          styles.input,
          {
            backgroundColor: 'lightgreen',
            marginHorizontal: 60,
          },
        ]}>
        <Text> {item} </Text>
      </View>
    );
  };
  const HandleClick = () => {
    if (!text) {
      return;
    }

    setList([...list, text]);
  };

  return (
    <SafeAreaView style={styles.container}>
      <Text>Test with Jest</Text>
      <FlatList
        testID="list"
        data={list}
        renderItem={renderList}
        keyExtractor={(_item, index) => index.toString()}
      />
      <TextInput
        onChangeText={setText}
        testID="Input-Area"
        placeholder="Add "
        style={styles.input}
      />
      <Button onPress={HandleClick} title="Add List" />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {flex: 1},
  input: {
    backgroundColor: 'lightgray',
    margin: 10,
    borderColor: 'black',
    borderRadius: 20,
    borderWidth: 1,
    padding: 12,
    marginHorizontal: 20,
  },
});

export default App;
