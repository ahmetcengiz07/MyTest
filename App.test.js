import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import App from './App';

test('should match will snapshot', () => {
  const comp = render(<App />);
  expect(comp).toMatchSnapshot();
});
test('should add item to list', () => {
  const comp = render(<App />);
  const input = comp.getByTestId('Input-Area');
  //test id si Input-Area olan komponenti getir

  fireEvent.changeText(input, 'test element');
  //inputa test element değerini yaz
  const submitButton = comp.getByTestId('button-touchable');
  fireEvent.press(submitButton);
  //submitbuttondaki ilgili test ıd inin olduğu komponenti getir ve ona basılıp basılmadığını test et

  const list = comp.getByTestId('list').props;
  expect(list.data.length).toBe(1);

  //submitButtonu 1 kez fireEvent ile çağırdığımız için datada 1 eleman mı oluşuyor diye kontrol ediyoruz
});
