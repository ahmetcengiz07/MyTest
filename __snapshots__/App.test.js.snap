// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match will snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <Text>
    Test with Jest
  </Text>
  <RCTScrollView
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    testID="list"
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
  <TextInput
    onChangeText={[Function]}
    placeholder="Add "
    style={
      {
        "backgroundColor": "lightgray",
        "borderColor": "black",
        "borderRadius": 20,
        "borderWidth": 1,
        "margin": 10,
        "marginHorizontal": 20,
        "padding": 12,
      }
    }
    testID="Input-Area"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "lightblue",
        "borderRadius": 50,
        "justifyContent": "center",
        "margin": 20,
        "opacity": 1,
        "padding": 20,
      }
    }
    testID="button-touchable"
  >
    <Text
      testID="button-title"
    >
      Add List
    </Text>
  </View>
</RCTSafeAreaView>
`;
